### PR TITLE
Ui overhaul rebase

### PR DESCRIPTION
**Pull request by Adrijaned**
Rebased onto latest develop, fixed to compile

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by Cervator** [comment]
This pull request **fixes 2 alerts** when merging 07c9b890115aa15155a793cfe72d8f0c95264d19 into aeea83999b43f235dd3829c9cf834176d330b6bb - [view on LGTM.com](https://lgtm.com/projects/g/MovingBlocks/DestinationSol/rev/pr-dc195933c7742c3f7515c57e713b405d641a6ff7)

**fixed alerts:**

* 1 for Dereferenced variable may be null
* 1 for Unused format argument

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: engine/build.gradle [hunk 1]
```diff
@@ -10,10 +30,10 @@ dependencies {
     compile "org.terasology:gestalt-module:6.0.0"
     compile "org.terasology:gestalt-asset-core:6.0.0"
     compile "org.terasology:gestalt-entity-system:6.0.0"
-
+    compile "com.github.everit-org.json-schema:org.everit.json.schema:1.9.2"
```
Comment (NicholasBatesNZ) [inline engine/build.gradle]:
```suggestion
    compile "com.github.everit-org.json-schema:org.everit.json.schema:1.11.1"
```

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Pull request merged**